lib/index.js

```diff
@@ -340,6 +350,51 @@ _.seq = function () {
     };
 };
 
+function promiseGenerator(promise) {
+    return _(function (push) {
+        promise.then(function (value) {
+                push(null, value);
+                return push(null, nil);
+            },
+            function (err) {
+                push(err);
+                return push(null, nil);
+            });
+    });
+}
+
+function iteratorGenerator(it) {
+    return _(function (push, next) {
+        var iterElem, iterErr;
+        try {
+            iterElem = it.next();
+        }
+        catch (err) {
+            iterErr = err;
+        }
+
+        if (iterErr) {
+            push(iterErr);
+            push(null, _.nil);
+        }
+        else if (iterElem.done) {
+            if (iterElem.value) {
```

Comment:
Sorry, yeah, that's a mistake. When` done` gets set to true `value` will always be undefined unless we're dealing with an iterator from a generator that uses the return keyword. I should be checking for the specific case of when `value` is not `undefined`. Not sure why there needs to be a test for value being false though?